src/libexpr/nixexpr.hh

```diff
@@ -79,16 +79,28 @@ struct Expr
 {
     virtual ~Expr() { };
     virtual void show(std::ostream & str) const;
+    virtual void showAsAterm(std::ostream & str) const;
+    virtual void showAsJson(std::ostream & str) const;
     virtual void bindVars(const StaticEnv & env);
     virtual void eval(EvalState & state, Env & env, Value & v);
     virtual Value * maybeThunk(EvalState & state, Env & env);
     virtual void setName(Symbol & name);
 };
 
+struct ExprAsAterm : Expr
+{
+    void show(std::ostream & str) const;
+};
+
+struct ExprAsJson : Expr
```

Comment:
What's the purpose of these classes?

Reply:
my attempt to solve this in the most lazy way:
by type-casting `Expr` to `ExprAsJson`, it should use the `showAsJson` methods